apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.Tests/V1/SpannerDateTests.cs

```diff
@@ -0,0 +1,193 @@
+            // input2 is greater than input1.
+            Assert.NotEqual(date1, date2);
+            Assert.False(date1 == date2);
+            Assert.False(date1.CompareTo(date2) == 0);
+            Assert.False(date2.CompareTo(date1) == 0);
+            Assert.False(date1.GetHashCode() == date2.GetHashCode());
+            Assert.True(date1 != date2);
+            Assert.True(!date1.Equals(date2));
+            Assert.True(!date2.Equals(date1));
+            Assert.True(date1 < date2);
+            Assert.True(date1 <= date2);
+            Assert.True(date1.CompareTo(date2) < 0);
+            Assert.True(date2 > date1);
+            Assert.True(date2 >= date1);
+            Assert.True(date2.CompareTo(date1) > 0);
+        }
+
+        [Theory]
+        [InlineData("0001-01-01", "0001-01-01")]
+        [InlineData("9999-12-31", "9999-12-31")]
+        [InlineData("2022-04-18", "2022-04-18")]
+        [InlineData("2000-02-28", "2000-02-28")]
+        public void Equality(string input1, string input2)
+        {
+            var date1 = SpannerDate.Parse(input1);
+            var date2 = SpannerDate.Parse(input2);
+            var hashCode1 = date1.GetHashCode();
+            var hashCode2 = date2.GetHashCode();
+            Assert.True(date1.Equals(date2));
+            Assert.True(date2.Equals(date1));
+            Assert.True(date1.CompareTo(date2) == 0);
+            Assert.True(date2.CompareTo(date1) == 0);
+            Assert.Equal(hashCode1, hashCode2);
+        }
+
+        [Theory]
+        [InlineData("9999-12-31", "9999-12-31")]
+        [InlineData("0001-01-01", "0001-01-01")]
+        [InlineData("2022-04-18", "2022-04-18")]
+        public void ToString_(string input, string expected)
+        {
+            SpannerDate spannerDate = SpannerDate.Parse(input);
+            Assert.Equal(expected, spannerDate.ToString());
+        }
+
+        [Theory]
+        [MemberData(nameof(FromDateTimeData))]
+        public void FromDateTime(DateTime input)
+        {
+            SpannerDate spannerDate = new SpannerDate(input);
+            Assert.Equal(input.Date.Year, spannerDate.Year);
+            Assert.Equal(input.Date.Month, spannerDate.Month);
+            Assert.Equal(input.Date.Day, spannerDate.Day);
+        }
+
+        [Theory]
+        [InlineData("9999-12-31")]
+        [InlineData("0001-01-01")]
+        [InlineData("2022-04-18")]
+        [InlineData("2000-12-31")]
+        public void ToDateTime(string input)
+        {
+            SpannerDate spannerDate = SpannerDate.Parse(input);
+            var expectedDateTime = DateTime.Parse(input, CultureInfo.InvariantCulture);
+            var result = spannerDate.ToDateTime();
+            Assert.Equal(expectedDateTime, result);
+            Assert.Equal(DateTimeKind.Unspecified, result.Kind);
+        }
+
+        [Fact]
+        public void Constants()
+        {
+            Assert.Equal(SpannerDate.Parse(MaxDate), SpannerDate.MaxDate);
```

Comment:
I'd just use `SpannerDate.FromDateTime(DateTime.MinValue)` and the same for max here.